bootstrapper/internal/joinclient/joinclient.go

```diff
@@ -188,8 +188,14 @@ func (c *JoinClient) Stop() {
 func (c *JoinClient) tryJoinWithAvailableServices() error {
 	ips, err := c.getControlPlaneIPs()
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get control plane IPs: %w", err)
+	}
+
+	ip, _, err := c.metadataAPI.GetLoadBalancerEndpoint(context.TODO())
```

Comment:
Consider moving the `ctx, cancel := c.timeoutCtx()` part out of `*JoinClient.getControlPlaneIPs`, and move it into `tryJoinWithAvailableServices`.
This would allow using the same context to fetch the control plane IPs, as well as the Load Balancer IPs.